argo-cd-apps/base/syncer.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: syncer
```

Comment:
syncer is very generic, proposing to change to `kcp-syncer` (everywhere, filenames, components/)